src/Common/Tippy.tsx

```diff
@@ -1,24 +1,22 @@
-import { useEffect, useRef, useState } from 'react'
-import TippyJS, { TippyProps as TippyJSProps } from '@tippyjs/react'
+import { useCallback, useState, Children, cloneElement } from 'react'
+import TippyJS from '@tippyjs/react'
+import { TippyProps } from './Types'
 
-// TODO: we should generalize this Tippy to work without sending in this truncateWidth
-const Tippy = ({ truncateWidth = 0, children, ...rest }: TippyJSProps & Record<'truncateWidth', number>) => {
-    // NOTE: if showOnTruncate is off then always showTippy
-    const [showTippy, setShowTippy] = useState(!!truncateWidth)
-    const ref = useRef(null)
+const Tippy = ({ showOnTruncate = false, children, ...rest }: TippyProps) => {
+    const [showTippy, setShowTippy] = useState(false)
 
-    useEffect(() => {
-        if (truncateWidth) {
-            setShowTippy(ref.current?.offsetWidth >= truncateWidth)
+    const refCallback = useCallback((node: HTMLDivElement) => {
+        if (node && showOnTruncate) {
+            setShowTippy(node.scrollWidth > node.clientWidth)
         }
-    }, [ref.current?.offsetWidth])
+    }, [])
```

Comment:
```suggestion
    }, [showOnTruncate])
```

Reply:
no need for passing showOnTruncate now since we can incorporate that into the logic that conditionally renders Tippy on child